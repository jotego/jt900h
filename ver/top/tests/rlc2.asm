    ; RLC tests from 1 to 15 shifts
    main section code
    org 0
    ld a,0xbf    ; common header
    ld w,0

    ld a,1
    ld xix,data
    ld xsp,stack

test_loop:
    call case
    inc 1,a
    cp a,0x10
    jp ne,test_loop

end_loop:
    ldf 0
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
case:
    ld d,0x81
    rlc A,d
    cp d,(xix+a)
    jp nz,bad_end
    inc 1,w
    ret
data:
    db 0x81,0x3,0x6,0xc,0x18,0x30,0x60,0xc0
    db 0x81,0x3,0x6,0xc,0x18,0x30,0x60,0xc0
    db 0x81,0x3,0x6,0xc,0x18,0x30,0x60,0xc0
    db 0x81,0x3,0x6,0xc,0x18,0x30,0x60,0xc0
    dw 0xffff,0xffff,0xffff,0xffff
stack:
    end