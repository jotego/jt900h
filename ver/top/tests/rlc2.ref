00000f10 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000c000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000046 - 00000000 - 00000000 - 0000006e
SR = f842
