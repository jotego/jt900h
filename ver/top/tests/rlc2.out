00000f10 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000c000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
0000003f - 00000000 - 00000000 - 00000067
SR = 42f8
